background.js

```diff
@@ -71,13 +72,20 @@ function bufferData(tabId, message) {
   }
 
   if (message.event === 'scope:new') {
-    tabData.scopes[message.data.child] = {
```

Comment:
It does. I just stored some properties as local var (`childScopeData = tabData.scopes[childId]`).